### PR TITLE
r/aws_apigatewayv2_integration: Add tls_config attribute

### PR DESCRIPTION
**Pull request by ewbankkit**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes https://github.com/terraform-providers/terraform-provider-aws/issues/12971.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_apigatewayv2_integration: Add `tls_config` attribute
```

Output from acceptance testing:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_basic
=== PAUSE TestAccAWSAPIGatewayV2Integration_basic
=== RUN   TestAccAWSAPIGatewayV2Integration_disappears
=== PAUSE TestAccAWSAPIGatewayV2Integration_disappears
=== RUN   TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_Lambda
=== PAUSE TestAccAWSAPIGatewayV2Integration_Lambda
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_basic
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_Lambda
=== CONT  TestAccAWSAPIGatewayV2Integration_disappears
--- PASS: TestAccAWSAPIGatewayV2Integration_disappears (24.16s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basic (27.95s)
--- PASS: TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp (45.83s)
--- PASS: TestAccAWSAPIGatewayV2Integration_Lambda (52.04s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkHttp (283.54s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket (690.60s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	690.661s
```

### CODE REVIEW

**Comment by AleksandarTokarev** [comment]
Any info when this is going to get released?

**Comment by ewbankkit** [comment]
#### TODO

Rebase after https://github.com/terraform-providers/terraform-provider-aws/pull/13062 is merged to incorporate the `passthrough_behavior` fix.

**Comment by ewbankkit** [comment]
Rebased to remove merge conflicts.
Re-ran acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_disappears
=== PAUSE TestAccAWSAPIGatewayV2Integration_disappears
=== RUN   TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_Lambda
=== PAUSE TestAccAWSAPIGatewayV2Integration_Lambda
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Integration_Lambda
=== CONT  TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_disappears
=== CONT  TestAccAWSAPIGatewayV2Integration_basicHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
--- PASS: TestAccAWSAPIGatewayV2Integration_disappears (12.66s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basicHttp (15.13s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basicWebSocket (16.42s)
--- PASS: TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp (39.44s)
--- PASS: TestAccAWSAPIGatewayV2Integration_Lambda (45.96s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkHttp (259.92s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket (736.72s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	736.773s
```

File: aws/resource_aws_apigatewayv2_integration_test.go [hunk 1]
```diff
@@ -445,17 +563,63 @@ resource "aws_apigatewayv2_integration" "test" {
...
+
+  connection_type      = "VPC_LINK"
+  connection_id        = "${aws_apigatewayv2_vpc_link.test.id}"
+  description          = "Test private integration updated"
+  integration_method   = "POST"
+  integration_uri      = "${aws_lb_listener.test.arn}"
+  timeout_milliseconds = 4999
+
+  tls_config {
+    server_name_to_verify = "www.example.org"
```
Comment (DrFaust92) [inline aws/resource_aws_apigatewayv2_integration_test.go:613]: no update to value from `testAccAWSAPIGatewayV2IntegrationConfig_vpcLinkHttp`?

File: aws/resource_aws_apigatewayv2_integration_test.go [hunk 2]
```diff
@@ -252,6 +256,74 @@ func TestAccAWSAPIGatewayV2Integration_VpcLink(t *testing.T) {
...
+					resource.TestCheckResourceAttr(resourceName, "passthrough_behavior", ""),
+					resource.TestCheckResourceAttr(resourceName, "payload_format_version", "1.0"),
+					resource.TestCheckResourceAttr(resourceName, "request_templates.%", "0"),
+					resource.TestCheckResourceAttr(resourceName, "template_selection_expression", ""),
+					resource.TestCheckResourceAttr(resourceName, "timeout_milliseconds", "5001"),
+					resource.TestCheckResourceAttr(resourceName, "tls_config.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tls_config.0.server_name_to_verify", "www.example.com"),
+				),
+			},
+			{
```
Comment (DrFaust92) [inline aws/resource_aws_apigatewayv2_integration_test.go:307]: add import test

File: aws/resource_aws_apigatewayv2_integration_test.go (hunk 1 above)
Comment (ewbankkit) [reply aws/resource_aws_apigatewayv2_integration_test.go:613]: Value in `testAccAWSAPIGatewayV2IntegrationConfig_vpcLinkHttp` is _www.example._***com***.

Comment (DrFaust92) [reply aws/resource_aws_apigatewayv2_integration_test.go:613]: Doh, missed that

File: aws/resource_aws_apigatewayv2_integration_test.go (hunk 2 above)
Comment (ewbankkit) [reply aws/resource_aws_apigatewayv2_integration_test.go:307]: Done.

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_VpcLinkHttp'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_VpcLinkHttp -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkHttp (343.05s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	343.109s
```

**Comment by breathingdust** [comment]
LGTM! 🚀 Thanks @ewbankkit 

Verified Acceptance Tests (Commercial)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_disappears
=== PAUSE TestAccAWSAPIGatewayV2Integration_disappears
=== RUN   TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_Lambda
=== PAUSE TestAccAWSAPIGatewayV2Integration_Lambda
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Integration_Lambda
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== CONT  TestAccAWSAPIGatewayV2Integration_basicHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_disappears
=== CONT  TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
--- PASS: TestAccAWSAPIGatewayV2Integration_basicHttp (13.85s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basicWebSocket (14.89s)
--- PASS: TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp (21.67s)
--- PASS: TestAccAWSAPIGatewayV2Integration_Lambda (33.69s)
--- PASS: TestAccAWSAPIGatewayV2Integration_disappears (40.55s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkHttp (259.93s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket (699.78s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	703.456s
```

Verified Acceptance Tests (GovCloud)

```console
make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSAPIGatewayV2Integration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAPIGatewayV2Integration_ -timeout 120m
=== RUN   TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_basicHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_basicHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_disappears
=== PAUSE TestAccAWSAPIGatewayV2Integration_disappears
=== RUN   TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== RUN   TestAccAWSAPIGatewayV2Integration_Lambda
=== PAUSE TestAccAWSAPIGatewayV2Integration_Lambda
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== RUN   TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== PAUSE TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_basicWebSocket
=== CONT  TestAccAWSAPIGatewayV2Integration_Lambda
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket
=== CONT  TestAccAWSAPIGatewayV2Integration_disappears
=== CONT  TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp
=== CONT  TestAccAWSAPIGatewayV2Integration_basicHttp
--- PASS: TestAccAWSAPIGatewayV2Integration_disappears (8.13s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basicHttp (9.95s)
--- PASS: TestAccAWSAPIGatewayV2Integration_basicWebSocket (10.13s)
--- PASS: TestAccAWSAPIGatewayV2Integration_IntegrationTypeHttp (13.96s)
--- PASS: TestAccAWSAPIGatewayV2Integration_Lambda (27.50s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkHttp (249.53s)
--- PASS: TestAccAWSAPIGatewayV2Integration_VpcLinkWebSocket (683.67s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	685.979s
```

**Pull request merged**